transmart-rest-api/grails-app/controllers/org/transmartproject/interceptors/QueryAuditInterceptor.groovy

```diff
@@ -0,0 +1,57 @@
+package org.transmartproject.interceptors
+
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class QueryAuditInterceptor extends AuditInterceptor {
+
+    QueryAuditInterceptor() {
+        match(controller: ~/query/,
+                action: ~/observations|counts|countsPerConcept|countsPerStudy|countsPerStudyAndConcept|aggregatesPerConcept|crosstable|table|supportedFields/)
+    }
+
+    boolean after() {
+        def ip = getIP()
+        Map<String, String> event = getEventMessage(ip)
+        report(event.event, event.eventMessage)
+    }
+
+    private Map<String, String> getEventMessage(ip) {
```

Comment:
They are really nice descriptive messages, but the audit logs are mainly meant for processing in other systems, not to be readable in a UI, so this seems a bit overkill.